From: David Rosca <nowrep@gmail.com>
Date: Fri, 25 Sep 2015 07:44:25 +0000
Subject: TagListJob: Emit error when failed to open database
X-Git-Url: http://quickgit.kde.org/?p=baloo.git&a=commitdiff&h=8258e544477ead37aea655eb0b7733f3ecaa93c9
---
TagListJob: Emit error when failed to open database

Emit job result when database can't be opened,
so TagListJob::exec() won't hang.

REVIEW: 125374
---


--- a/src/lib/taglistjob.cpp
+++ b/src/lib/taglistjob.cpp
@@ -47,7 +47,9 @@
 {
     Database *db = globalDatabaseInstance();
     if (!db->open(Database::OpenDatabase)) {
-        qCritical() << "Failed to open the database";
+        setError(UserDefinedError);
+        setErrorText(QStringLiteral("Failed to open the database"));
+        emitResult();
         return;
     }
 


